### ISSUE TITLE
Explicit default copy assignment operator for boost::bimaps::detail::map_view_iterator?

### ISSUE THREAD

**Issue by garethsb-sony** [boostorg/bimap#22]
Adding an explicit default copy assignment operator to `boost::bimaps::detail::map_view_iterator` suppresses a warning from GCC 9.1+ about https://en.cppreference.com/w/cpp/language/copy_assignment#Implicitly-defined_copy_assignment_operator due to the explicit copy constructor.

```patch
+    map_view_iterator& operator=(map_view_iterator const& iter) = default;
```

Is that the right thing to do?

**Comment by glenfe** [comment]
Bimap needs to support C++03 so that would have to be added conditionally only when the newer compilers are detected. But I don't think that's the best solution. Instead we should get rid of the explicit copy constructor, so that both copy constructor and assignment operators are implicitly provided.

**Comment by glenfe** [comment]
Addressed in 6b4d000a4fb38978c87a48c7c22344e61a33290f.
